pkg/fake/iamapi.go

```diff
@@ -39,126 +40,169 @@ type IAMAPIBehavior struct {
 	AddRoleToInstanceProfileBehavior      MockedFunction[iam.AddRoleToInstanceProfileInput, iam.AddRoleToInstanceProfileOutput]
 	TagInstanceProfileBehavior            MockedFunction[iam.TagInstanceProfileInput, iam.TagInstanceProfileOutput]
 	RemoveRoleFromInstanceProfileBehavior MockedFunction[iam.RemoveRoleFromInstanceProfileInput, iam.RemoveRoleFromInstanceProfileOutput]
+	UntagInstanceProfileBehavior          MockedFunction[iam.UntagInstanceProfileInput, iam.UntagInstanceProfileOutput]
 }
 
 type IAMAPI struct {
 	sync.Mutex
 
-	iamiface.IAMAPI
+	sdk.IAMAPI
 	IAMAPIBehavior
 
-	InstanceProfiles map[string]*iam.InstanceProfile
+	InstanceProfiles map[string]*iamtypes.InstanceProfile
 }
 
 func NewIAMAPI() *IAMAPI {
-	return &IAMAPI{InstanceProfiles: map[string]*iam.InstanceProfile{}}
+	return &IAMAPI{InstanceProfiles: map[string]*iamtypes.InstanceProfile{}}
 }
 
 // Reset must be called between tests otherwise tests will pollute
 // each other.
+func toSliceOfValues[T any](value []T) []T {
```

Comment:
Can't we use `aws.StringSlice`? https://github.com/aws/aws-sdk-go-v2/blob/c5e364e134b6207b81c762ad18f41d09ec531f21/aws/from_ptr.go#L62

Reply:
StringSlice returns []*string but it needs []string